### ISSUE TITLE
fails to build on Windows

### ISSUE THREAD

**Issue by astraw** [jerry73204/apriltag-sys#6]
I cannot get this to build on Windows.

Here is what I did:

Install prerequisites:

- [cmake](https://cmake.org/download/).
- Python [from the Microsoft Store](https://go.microsoft.com/fwlink?linkID=2082640)
- numpy with `pip install numpy`
- vcpkg from [https://github.com/Microsoft/vcpkg](https://github.com/Microsoft/vcpkg)
- pthreads vcpkg with ` .\vcpkg install pthreads`

```text
set DCMAKE_TOOLCHAIN_FILE=C:/Users/drand/Documents/src/vcpkg/scripts/buildsystems/vcpkg.cmake
set APRILTAG_SYS_METHOD=cmake,dynamic
cargo test
```

This is the result

```
    Updating crates.io index
   Compiling jobserver v0.1.21
   Compiling lazy_static v1.4.0
   Compiling glob v0.3.0
   Compiling pkg-config v0.3.17
   Compiling cc v1.0.50
   Compiling cmake v0.1.42
   Compiling apriltag-sys v0.1.3 (C:\Users\drand\Documents\src\apriltag-sys)
error: failed to run custom build command for `apriltag-sys v0.1.3 (C:\Users\drand\Documents\src\apriltag-sys)`

Caused by:
  process didn't exit successfully: `C:\Users\drand\Documents\src\apriltag-sys\target\debug\build\apriltag-sys-5303f535d37770f0\build-script-build` (exit code: 101)
--- stdout
cargo:rerun-if-changed=wrapper.h
cargo:rerun-if-env-changed=APRILTAG_SRC
cargo:rerun-if-env-changed=APRILTAG_SYS_METHOD
cargo:rustc-link-lib=apriltag
running: "cmake" "C:\\Users\\drand\\Documents\\src\\apriltag-sys\\apriltag-src" "-G" "Visual Studio 16 2019" "-Thost=x64" "-Ax64" "-DCMAKE_INSTALL_PREFIX=C:\\Users\\drand\\Documents\\src\\apriltag-sys\\target\\debug\\build\\apriltag-sys-08db87b3a475d233\\out\\cmake_build" "-DCMAKE_C_FLAGS= -nologo -MD -Brepro" "-DCMAKE_C_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_CXX_FLAGS= -nologo -MD -Brepro" "-DCMAKE_CXX_FLAGS_DEBUG= -nologo -MD -Brepro" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_TOOLCHAIN_FILE=C:/Users/drand/Documents/src/vcpkg/scripts/buildsystems/vcpkg.cmake"
-- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.18363.
-- The C compiler identification is MSVC 19.24.28319.0
-- The CXX compiler identification is MSVC 19.24.28319.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.24.28314/bin/Hostx64/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.24.28314/bin/Hostx64/x64/cl.exe - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.24.28314/bin/Hostx64/x64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.24.28314/bin/Hostx64/x64/cl.exe - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PythonLibs: C:/Program Files/WindowsApps/PythonSoftwareFoundation.Python.3.8_3.8.752.0_x64__qbz5n2kfra8p0/libs/python38.lib (found version "3.8.2")
-- Configuring done
-- Generating done

--- stderr
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
PTHREAD_LIBRARIES
    linked by target "apriltag" in directory C:/Users/drand/Documents/src/apriltag-sys/apriltag-src

CMake Generate step failed.  Build files cannot be regenerated correctly.
thread 'main' panicked at '
command did not execute successfully, got: exit code: 1

build script failed, must exit now', C:\Users\drand\.cargo\registry\src\github.com-1ecc6299db9ec823\cmake-0.1.42\src\lib.rs:861:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

It looks like [it is possible to build apriltags on Windows](https://github.com/AprilRobotics/apriltag/pull/27), but so far I did not succeed.

**Comment by jerry73204** [comment]
The error is due to running `pkg-config` on Windows, which will fail apparently. We may count on [vcpkg](https://docs.rs/vcpkg/) to make it work. [opencv-rust](https://github.com/twistedfall/opencv-rust/blob/1ba990a7127ec0518d22e016c4d83d34c0a5059e/build.rs#L396-L413) also work it out already. I may give it a try.

**Comment by astraw** [comment]
closed by #8
